angel_system/tcn_hpl/predict.py

```diff
@@ -226,6 +229,88 @@ def predict(
     return torch.softmax(logits[-1, :, :, -1], dim=1)[0]
 
 
+@dataclass
+class ResultElement:
+    """
+    Single activity classification result element for later inclusion in an
+    output COCO file.
+    """
+
+    # Class index of the single most confident activity class.
+    pred_index: int
+    # Vector of confidence values for all activity classes.
+    conf_vec: List[float]
+
+
+class ResultsCollector:
+    """
+    Utility class to collect TCN-HPL prediction results, encoding the ability
+    to output the results to a COCO file.
+
+    This is a beginning to generalizing results coco file aggregation and
+    output, based on output from `python-tpl/TCN_HPL/tcn_hpl/models/ptg_module.py`
+    in `on_validation_epoch_end` and `on_test_epoch_end` methods.
+
+    TODO: Promote into TCN_HPL repo?
```

Comment:
This makes sense to me in the TCN_HPL repo, otherwise LGTM

Reply:
I didn't put it there because I didn't see a direct path to making use of it within existing functionality there yet, as there would be debugging to do there, so its just used in the angel_system content right now.